src/course-outline/CourseOutline.jsx

```diff
@@ -125,6 +126,27 @@ const CourseOutline = ({ courseId }) => {
     });
   };
 
+  const setUnit = (sectionIndex, subsectionIndex) => (updatedUnits) => {
+    const section = { ...sections[sectionIndex] };
+    section.childInfo = { ...section.childInfo };
+
+    const subsection = { ...section.childInfo.children[subsectionIndex] };
+    subsection.childInfo = { ...subsection.childInfo };
+    subsection.childInfo.children = updatedUnits();
+
+    const updatedSubsection = [...subsection.childInfo.children];
+    updatedSubsection[subsectionIndex] = subsection;
+    section.childInfo.children = updatedSubsection;
```

Comment:
There is a small issue with this code, it is setting units as children of sections so whenever you drag-n-drop a unit, the corresponding section's children are briefly updated to units and then correctly set once the request to server completes. Please update it as below to fix the issue.

```suggestion
    const updatedSubsections = [...section.childInfo.children];
    updatedSubsections[subsectionIndex] = subsection;
    section.childInfo.children = updatedSubsections;
```